### ISSUE TITLE
nushell doesn't execute items in path in order

### ISSUE THREAD

**Issue by fdncred** [nushell/nushell#5070]
### Describe the bug

I intentionally put sqlite3 as the first element in my path. However, after restarting nushell, it still doesn't execute this one first. Instead it finds the one in the 10th slot, /usr/bin, and executes it.

```
> which sqlite3
╭───┬─────────┬──────────────────┬──────────╮
│ # │   arg   │       path       │ built-in │
├───┼─────────┼──────────────────┼──────────┤
│ 0 │ sqlite3 │ /usr/bin/sqlite3 │ false    │
├───┼─────────┼──────────────────┼──────────┤
│ # │   arg   │       path       │ built-in │
╰───┴─────────┴──────────────────┴──────────╯
> $env.PATH
╭────┬────────────────────────────────────────────────╮
│  0 │ /opt/homebrew/Cellar/sqlite/3.38.2/bin/sqlite3 │
│  1 │ /Users/fdncred/.cargo/bin                      │
│  2 │ /opt/homebrew/opt/coreutils/libexec/gnubin     │
│  3 │ /Users/fdncred/.cargo/bin                      │
│  4 │ /opt/homebrew/bin                              │
│  5 │ /opt/homebrew/sbin                             │
│  6 │ /usr/local/bin                                 │
│  7 │ /usr/bin                                       │
│  8 │ /bin                                           │
│  9 │ /usr/sbin                                      │
│ 10 │ /sbin                                          │
╰────┴────────────────────────────────────────────────╯
```
In fact, even though it's in my path, nushell doesn't see it.
```
> which sqlite3 -a 
╭───┬─────────┬──────────────────┬──────────╮
│ # │   arg   │       path       │ built-in │
├───┼─────────┼──────────────────┼──────────┤
│ 0 │ sqlite3 │ /usr/bin/sqlite3 │ false    │
├───┼─────────┼──────────────────┼──────────┤
│ # │   arg   │       path       │ built-in │
╰───┴─────────┴──────────────────┴──────────╯
```

### How to reproduce

I put this in my config.nu to ensure that this folder was searched first.
`let-env PATH = ($env.PATH | prepend '/opt/homebrew/Cellar/sqlite/3.38.2/bin/sqlite3')`

### Expected behavior

Look through the path in order already!

### Configuration

|key|value|
|-|-|
|version|0.60.1|
|branch|main|
|short_commit|e2d24c59|
|commit_hash|e2d24c59562985b2ade539352b0bc0f009f375c7|
|commit_date|2022-03-30 18:37:31 +00:00|
|build_os|macos-aarch64|
|rust_version|rustc 1.59.0 (9d1b2106e 2022-02-23)|
|rust_channel|stable-aarch64-apple-darwin|
|cargo_version|cargo 1.59.0 (49d8809dc 2022-02-10)|
|pkg_version|0.60.1|
|build_time|2022-03-30 19:34:05 -05:00|
|build_rust_channel|debug|
|features|dataframe, default, trash, which, zip|
|installed_plugins|gstat, nu-example-1, nu-example-2, nu-example-3, query, query json, query web, query xml|

**Comment by uasi** [comment]
> I put this in my config.nu to ensure that this folder was searched first.

Isn't `/opt/homebrew/Cellar/sqlite/3.38.2/bin/sqlite3` a file?

**Comment by fdncred** [comment]
yes, it is a file, you're right. Also the path is wrong now. let me test again.

yay! I'm an idiot, again!

I changed it to this `let-env PATH = ($env.PATH | prepend '/opt/homebrew/opt/sqlite/bin')` after I ran `brew link sqlite` so now it's linked in that path. apparently you're not supposed to link to what I had in the initial post/whine.

Thanks for pointing out I was wrong. TBH, I'd rather be wrong about this than have some weird bug to track down. I'm glad it was me and not my friend nushell.
